tests/pytest/test_utils.py

```diff
@@ -0,0 +1,38 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+import pytest
+
+import cocotb.utils
+
+
+class TestHexDump:
+    def test_int_illegal(dut):
+        # this used to be legal, but deliberately is no longer
+        with pytest.raises(TypeError):
+            cocotb.utils.hexdump(1)
```

Comment:
Thanks to my delay in responding to suggestions, the tests can now be done in pytest